praw/models/reddit/collections.py

```diff
@@ -143,21 +145,30 @@ def __setattr__(self, attribute, value):
             value = self._reddit._objector.objectify(value)
         super(Collection, self).__setattr__(attribute, value)
 
+    def _fetch_info(self):
```

Comment:
Can you keep these methods definitions sorted please?

Reply:
I would like to keep it the way it is since it reads better when read linearly.

Lexicological ordering here makes the logic less easy to follow, and in general provides fewer readability guarantees. Prearranged orderings on method names means that a name change would require having to reorder methods around. It’s an arbitrary change to have to make, and the resulting order is virtually random in term of how methods relate.

Also consider that method order can be influenced by finding alternative names, rather than sticking to more accurate ones. For instance, in this case, I could have named “`_fetch_data`” as “`_fetch_json`” and the order would then be correct, but then I’d find myself in a situation having to decide whether to optimise for readability by having a nice method ordering, or by having a good ordering but a less suitable method name.

When you enforce lexicological sorting you lose the ability to express meaning through order, and that can be useful in many cases.

Reply:
🤷‍♂ It's clear you have some strong opinions on ordering, and you make some good points that absolutely make sense. However, with a project having many contributors methods, parameters, etc typically get added in an arbitrary order. Imposing a lexicographic order means there's exactly one place to put things. I've personally never considered renaming something just to enable things to be ordered the way I want it. That said, I'm sure some people would.

I'm not going to require you to put them in order, but it is part of the project style guide to keep methods sorted: https://praw.readthedocs.io/en/latest/package_info/contributing.html#code

So please don't be bummed if the methods get reordered later.

Reply:
I feel that these kinds of details are just there to make minute decisions easier. And there’s comfort in doing so. But I don’t think the system works well in many cases, especially when it’s really obvious when methods like `__eq__`, `__ne__`, `__lt__`, `__gt__`, `__ge__`, `__le__` should be together with `__eq__` at the top (ignoring the coincidence).

I’m not against one way of writing things though, and even though there isn’t a hard and fast rule to follow, I believe that there still can be one and only one preferable way of writing things, if not obvious to others.

I understand that ordering can be subjective in a project though. If I haven’t convinced you and you require the ordering as per the guideline point, just say so.

Reply:
> you require the ordering as per the guideline point, just say so.

You're a member of this project now. So you should feel free to write it as you're comfortable with. I will continue to suggest people put things in lexicographic order, but you have good reasons for not doing so. Thanks for the discussion.